ProxyStarcraft/Unit.cs

```diff
@@ -36,6 +36,15 @@ public Unit(Proto.Unit unit, Translator translator)
         public bool IsBuildingSomething => translator.IsBuildingSomething(this.Raw);
 
         public bool HasBuff(BuffType buff) => this.Raw.BuffIds.Any(rawBuffId => translator.MatchingBuffIds(buff).Any(matching => matching == rawBuffId));
+        
+        /// <summary>
+        /// Sets a <seealso cref="BuffType"/> for a unit's game state (so that other computation within the same frame doesn't attempt to do something else with the unit)
+        /// </summary>
+        /// <param name="buff"></param>
+        /// <remarks>This will technically add a bunch of buffs that an entity "can't" have, but are effectively all identical. 
+        /// (e.g. "Banshee Cloak", "Ghost Cloak", "Dark Templar Cloak", and "Mothership Field Cloak")
+        /// </remarks>
+        public void SetBuff(BuffType buff) => this.Raw.BuffIds.AddRange(translator.MatchingBuffIds(buff));
```

Comment:
Avoid putting translator stuff or proto stuff in these classes where possible, e.g., just store at most the two cloak types (would maybe prefer they be a single cloak type, although I suppose there are mechanical implications of cloak field vs. cloak).